libs/linkBankOperations.js

```diff
@@ -102,31 +76,72 @@ const findReimbursedOperation = (bill, operations, options) => {
 }
 
 class Linker {
-  constructor (cozy) {
-    this.cozy = cozy
+  constructor (cozyClient) {
+    this.cozyClient = cozyClient
   }
 
   fetchNeighboringOperations (bill, options) {
-    if (typeof options.minDateDelta === 'undefined' || typeof options.maxDateDelta === 'undefined') {
-      return Promise.reject(new Error('Must have options.{min,max}DateDelta'))
+    const createDateSelector = () => {
+      const date = new Date(bill.originalDate || bill.date)
+      const startDate = moment(date).subtract(options.minDateDelta, 'days')
+      const endDate = moment(date).add(options.maxDateDelta, 'days')
+
+      // Get the operations corresponding to the date interval around the date of the bill
+      return {
+        $gt: `${startDate.format('YYYY-MM-DDT00:00:00.000')}Z`,
+        $lt: `${endDate.format('YYYY-MM-DDT00:00:00.000')}Z`
+      }
+    }
+
+    const createAmountSelector = () => {
+      const amount = bill.isRefund ? bill.amount : -bill.amount
+      const min = amount - options.minAmountDelta
+      const max = amount + options.maxAmountDelta
+
+      return {
+        $gt: min,
+        $lt: max
+      }
+    }
+
+    let operations = []
+    const dateSelector = createDateSelector()
+    const amountSelector = createAmountSelector()
+
+    const limit = 100
+    const getOptions = ids => {
+      const options = {
+          selector: {
+            date: dateSelector,
+            amount: amountSelector
+          },
+          sort: [{date: 'desc'}, {amount: 'desc'}],
```

Comment:
I see above "it is not possible to sort with 2 parameters", is it related ?

Reply:
pas vraiment la c'est les priorités pour trier les retours de couchdb. Donc le paramètre date est d'abord utilisé puis pour les résultats ayant la même date, on trie sur le montant.

Nous on voulait pouvoir trier avec les 2 paramètres directement et pas l'un après l'autre ;)